### PR TITLE
more spec for consensus faults

### PR DESCRIPTION
**Pull request by sternhenri**
more specifics and a fix for parent-grinding

### CODE REVIEW

**Comment by anorth** [comment]
Thanks. Is anything here different from go-filecoin's [implementation](https://github.com/filecoin-project/go-filecoin/blob/d75e700d2b5b116a794c3da672d6379bcb24b803/internal/pkg/slashing/check.go#L40)?

**Comment by anorth** [comment]
I see https://github.com/filecoin-project/go-filecoin/pull/3983

File: src/algorithms/expected_consensus/_index.md [hunk 1]
```diff
@@ -190,15 +190,26 @@ EC enforces a version of soft finality whereby all miners at round N will reject
...
+- (3) `parent-grinding fault`: one block's parent is a Tipset that provably should have included a given block but does not. While it cannot be proven that a missing block was willfully omitted in general (i.e. network latency could simply mean the miner did not receive a particular block), it can when a miner has successfully mined a block two epochs in a row and omitted one. That is, this condition should be evoked when a miner omits their own prior block.
+  - Specifically, this can be proven with a "witness" block, that is by submitting blocks A, B, C where C is B's parent and A's sibling but A is not B's parent.
+  - `!blockB.Parents.Include(blockA) && blockB.Parents.Include(blockC) && blockA.Parents == blockC.Parents && blockA.Height == blockC.Height`
```
Comment (anorth) [inline src/algorithms/expected_consensus/_index.md:211]: `blockA.Height == blockC.Height` is a double-fork mining fault already

Comment (sternhenri) [reply src/algorithms/expected_consensus/_index.md:211]: Detailed in https://github.com/filecoin-project/go-filecoin/pull/3983#discussion_r407541095, but not if `blockA.Parents != blockC.Parents`.

Comment (anorth) [reply src/algorithms/expected_consensus/_index.md:211]: Ack. Could you update the diagram here (linked below) to match please?

**Pull request merged**